tests/typeclass-tests.dx

```diff
@@ -190,15 +190,18 @@ instance {a} AutoQuant (MyPairOfXs a)
 
 -- Should be able to quantify nested constraints
 def q5 {a} (x : MyPairOfXs (MyPairOfXs a)) : Int = 0
-:t q5
-> ((a:Type) ?-> (X a) ?=> (X (a & a)) ?=> ((a & a) & (a & a)) -> Int32)
+
+-- disabling because the order of constraints changes depending on the names in
+-- scope
+-- :t q5
+-- > ((a:Type) ?-> (X (a & a)) ?=> (X a) ?=> ((a & a) & (a & a)) -> Int32)
```

Comment:
I don't follow. Checking the type will fail depending on what's in the cache.

Reply:
_Checking_ the type will fail? I understood this comment as "depending on how you edit this file, this type might print differently". Why is also surprising - why do we have this order dependence?

Reply:
The order dependence is because the constraints come from auto-quantification, so the order is arbitrary. In the implementation we collect a set of constraints and then turn it into a list. Docstring for `toList` is here: https://hackage.haskell.org/package/unordered-containers-0.2.16.0/docs/Data-HashMap-Strict.html#v:toList . 

"O(n) Return a list of this map's elements. The list is produced lazily. The order of its elements is unspecified."